packages/idyll-components/src/switch.js

```diff
@@ -0,0 +1,34 @@
+const React = require('react');
+
+class Switch extends React.Component {
+  render() {
+    const { idyll, hasError, updateProps, value, ...props } = this.props;
+
+    if (props.children) {
+      const filterFn = value => ({ props }) =>
+        props.children[0].props.test === value;
+      const matchCase = filterFn(value);
```

Comment:
Can you do the filter using the `filterChildren` function from https://github.com/idyll-lang/idyll-component-children? This will make it more future-proof in case we change how Idyll wraps components

Reply:
Updated this to use idyll-component-children.